src/olympia/reviewers/serializers.py

```diff
@@ -123,27 +134,28 @@ def _fetch_entries():
 
         return self._entries
 
-    def is_binary(self, filepath, mimetype, blob):
-        """
-        Using filepath, mimetype and in-memory buffer to determine if a file
-        can be shown in HTML or not.
+    def get_entry_mime_type(self, entry, blob):
+        """Returns the mimetype and type category.
+
+        The type category can be ``image``, ``directory``, ``text`` or
+        ``binary``.
         """
-        # Re-use the denied data from amo-validator to spot binaries.
-        ext = os.path.splitext(filepath)[1][1:]
-        if ext in denied_extensions:
-            return True
+        if entry.type == 'tree':
+            return 'application/octet-stream', 'directory'
 
-        bytes_ = tuple(bytearray(memoryview(blob)[:4]))
+        # Hardcoding the maximum amount of bytes to read here
+        # until https://github.com/ahupp/python-magic/commit/50e8c856
+        # lands in a release and we can read that value from libmagic
+        bytes_ = io.BytesIO(memoryview(blob)).read(1048576)
```

Comment:
I don't get this. Isn't that value the default ? Is something else altering the max value if we don't set it ?

Reply:
Updated the comment, the value is the default, correct, but I have to wrap this to not exhaust/read the whole blob into memory again (e.g for large files).